src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -1202,8 +1202,11 @@ private static class EmittingVisitor
                             var newExpr = (NewExpression)expr;
                             var argExprs = newExpr.Arguments;
                             for (var i = 0; i < argExprs.Count; i++)
-                                if (!TryEmit(argExprs[i], paramExprs, il, ref closure, parent, i))
+                            {
+                                var idx = argExprs[i].Type.IsByRef ? i : -1;
+                                if (!TryEmit(argExprs[i], paramExprs, il, ref closure, parent, idx))
                                     return false;
+                            }

```

Comment:
@dzmitry-lahoda again, is this correct fix for new with byref?

Reply:
yep. Originally I missed that in .NET there is more then one way to call `method` (usual, and via delegate and via ctor).